python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -209,3 +383,201 @@ def proc_to_examine():
+            pass
+        return True
+
+    loc_limit="25 KB"
+    if (MACOS):
+        loc_limit = "35 KB"
+    @MEMRAY_TESTS_MARK
+    @pytest.mark.limit_leaks(location_limit=loc_limit, filter_fn=is_relevant)
+    def test_mem_leak_querybuilder_read_memray(library_with_symbol):
+        """
+            Test to capture memory leaks >= of specified number
+
+            NOTE: we could filter out not meaningful for us stackframes
+            in future if something outside of us start to leak using
+            the argument "filter_fn" - just add to the filter function
+            what we must exclude from calculation
+        """
+        (lib, df) = library_with_symbol
+        mem_query(lib, df)
+
+
+    @MEMRAY_TESTS_MARK
+    @pytest.mark.limit_leaks(location_limit="25 KB", filter_fn=is_relevant)
+    def test_mem_leak_querybuilder_read_batch_memray(library_with_symbol):
+        """
+            Test to capture memory leaks >= of specified number
+
+            NOTE: we could filter out not meaningful for us stackframes
+            in future if something outside of us start to leak using
+            the argument "filter_fn" - just add to the filter function
+            what we must exclude from calculation
+        """
+        (lib, df) = library_with_symbol
+        mem_query(lib, df, read_batch=True)
+
+
+    @MEMRAY_TESTS_MARK
+    @pytest.mark.limit_memory("490 MB")
+    def test_mem_limit_querybuilder_read_memray(library_with_symbol):
+        """
+            The fact that we do not leak memory does not mean that we
+            are efficient on memory usage. This test captures the memory usage
+            and limits it, so that we do not go over it (unless fro good reason)
+            Thus if the test fails then perhaps we are using now more memory than
+            in the past
+        """
+        (lib, df) = library_with_symbol
+        mem_query(lib, df)
+
+    @MEMRAY_TESTS_MARK
+    @pytest.mark.limit_memory("490 MB")
+    def test_mem_limit_querybuilder_read_batch_memray(library_with_symbol):
+        """
+            The fact that we do not leak memory does not mean that we
+            are efficient on memory usage. This test captures the memory usage
+            and limits it, so that we do not go over it (unless fro good reason)
+            Thus if the test fails then perhaps we are using now more memory than
+            in the past
+        """
+        (lib, df) = library_with_symbol
+        mem_query(lib, df, True)
+
+
+    @pytest.fixture
+    def library_with_big_symbol(arctic_library_lmdb) -> Generator[Library, None, None]:
+        """
+            As memray instruments memory, we need to take out 
+            everything not relevant from mem leak measurement out of 
+            test, so it works as less as possible
+        """
+        lib: Library = arctic_library_lmdb
+        df = generate_big_dataframe(300000)
+        lib.write(symbol, df)
+        del df
+        yield lib
```

Comment:
Prefer using `get_sample_dataframe` everywhere than these custom fixtures

Reply:
generate_big_dataframe(100000)
Generation took : 6.429168701171875
df = get_sample_dataframe(16000000) #This is equivalent to generate_big_dataframe(100000)
Time:  49.78162121772766

Quite huge difference. 8x times and for larger DFs would be more. Thus for mem leaks etc time wize generation should be preferred